src/d3a/d3a_core/live_events.py

```diff
@@ -50,14 +50,15 @@ def __init__(self, area_uuid, area_params):
     def apply(self, area):
         if area.uuid != self.area_uuid:
             return False
-        self.sanitize_live_event_paramters_for_strategy_update()
         area_type = self.area_params.pop("type", None)
         if area_type is not None:
+            self.sanitize_live_event_paramters_for_strategy_update()
```

Comment:
This implies that the number_of_clones and the energy_rate parameters do not get removed if the area is a market and not a strategy. Is this really intended ? Moreover, the case that the "type" == "Area" is not covered (if the logic is the same, the `sanitize_live_event_paramters_for_strategy_update` call should be avoided too).

Reply:
I agree with the latter.